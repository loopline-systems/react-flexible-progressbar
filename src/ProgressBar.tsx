import React from 'react';

interface Props {
  width?: number | string;
  height?: number | string;
  margin?: number | string;
  progress?: number;
  progressTxt?: string;
  showPercentage?: boolean;
}

const ProgressBar = ({
  width,
  height = 10,
  margin,
  progress = 0,
  progressTxt = 'Progress:',
  showPercentage,
}: Props) => (
  <div className="react-flexible-progressbar" style={{ width, margin }}>
    <div className="rf-progressbar" style={{ height }}>
      <div style={{ width: `${progress}%` }} />
    </div>
    {showPercentage && (
      <div className="rf-progress-txt">{progressTxt} {`${progress}%`}</div>
    )}
  </div>
);

export default ProgressBar;
