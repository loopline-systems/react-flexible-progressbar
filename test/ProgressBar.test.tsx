import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProgressBar from '../src/ProgressBar';
import toJson from 'enzyme-to-json';

describe('ProgressBar', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  test('renders correctly', () => {
    const component = shallow(
      <ProgressBar width={200} progress={10} showPercentage />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  test('renders correctly without percentage text', () => {
    const component = shallow(
      <ProgressBar width={200} progress={10} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
