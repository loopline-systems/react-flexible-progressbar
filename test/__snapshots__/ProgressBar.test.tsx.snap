// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders correctly 1`] = `
<div
  className="react-flexible-progressbar"
  style={
    Object {
      "margin": undefined,
      "width": 200,
    }
  }
>
  <div
    className="rf-progressbar"
    style={
      Object {
        "height": 10,
      }
    }
  >
    <div
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
  <div
    className="rf-progress-txt"
  >
    Progress:
     
    10%
  </div>
</div>
`;

exports[`ProgressBar renders correctly without percentage text 1`] = `
<div
  className="react-flexible-progressbar"
  style={
    Object {
      "margin": undefined,
      "width": 200,
    }
  }
>
  <div
    className="rf-progressbar"
    style={
      Object {
        "height": 10,
      }
    }
  >
    <div
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;
